src/matrix.js

```diff
@@ -874,6 +874,26 @@ export class AbstractMatrix {
     return result;
   }
 
+  mpow(scalar) {
+    if (!this.isSquare()) {
+      throw new RangeError('Matrix must be square');
+    }
+    if (!Number.isInteger(scalar) || scalar < 0) {
+      throw new RangeError('Exponent must be a non-negative integer');
+    }
+    // Russian Peasant exponentiation, i.e. exponentiation by squaring
+    let result = Matrix.eye(this.rows);
+    let bb = this;
+    // Note: Don't bit shift. In JS, that would truncate at 32 bits
+    for (let e = scalar; e > 1; e /= 2) {
+      if (e & (1 !== 0)) {
```

Comment:
This looks like a bug. `(1 !== 0)` is a constant `true`.

Reply:
Can you please add a more complex test case that would fail because of this bug?

Reply:
Funnily enough, this does not lead to different behavior. `e & true` coerces `true` to 1 and winds up extracting the same bit!

Anyway, improved tests a bit more, including a high integer exponent.